tsconfig.json

```diff
@@ -5,8 +5,8 @@
     "declaration": true,
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
-    "jsx": "react",
-    "lib": ["dom", "es6", "dom.iterable"],
+    "jsx": "react-jsx",
+    "lib": ["DOM", "ES6", "DOM.Iterable", "ES2021"],
```

Comment:
FWIW, here's where I've ended up in frontend-base:

```
"jsx": "react-jsx",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
```

Admittedly I don't 100% comprehend exactly including things here gets you.  Do you really need ES6 _and_ ES2021?

Reply:
> Admittedly I don't 100% comprehend exactly including things here gets you. Do you really need ES6 and ES2021?

On a quick test locally, it does appear that specifying a later spec, you don't need the older ones. E.g., `ESNext` also resolves the `.replaceAll()` type issue I'm seeing, without explicitly needing to specify `ES2021`. 

So yeah, probably don't need "ES6" _and_ "ES2021". Any objection if I change to "esnext" to match what you have in frontend-base? I don't think it has too much downside? 🤔

Reply:
Related, I also wonder if we even need "DOM.Iterable" if we also include all of "DOM"?

Reply:
From what I'm reading, adding stuff to lib just changes what type definitions are available in your source code, so a more permissive list here is kind of okay.  The 'target' determines what features get transpiled down ('down-leveled' in the TS documentation).  So since our target is ES6, we should be good.

I wasn't quite clear on DOM/DOM.Iterable.  I do know that create-react-app generates a tsconfig.json with both. :)  (In fact, it matches what I put in frontend-base, which is why I did it...)

Reply:
For reference, this is a `create-react-app` tsconfig.json file:

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src"
  ]
}```

Reply:
[inform] I've updated to match the `lib` as frontend-base / CRA configures it (leaving in the `DOM` _and_ `DOM.iterable`).